src/test/java/com/google/sps/ProductPhotoDetectorTest.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.ProductDetectionData;
+
+import com.google.common.collect.ImmutableList;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** 
+ * Tests the behaviour of {@link ProductPhotoDetector#buildShoppingQuery(byte[])}.
+ */
+@RunWith(JUnit4.class)
+public final class ProductPhotoDetectorTest {
+
+  private static final byte[] IMAGE_BYTES = new byte[1];
+
+  private FakeProductDetectionAPIImpl fakeProductDetection;
+
+  @Before
+  public void setUp() {
+    fakeProductDetection = new FakeProductDetectionAPIImpl();
+  }
+
+  /** Sets product detection results and initializes ProductPhotoDetector instance. */
+  private ProductPhotoDetector initProductPhotoDetector(ImmutableList<String> labels, 
+      ImmutableList<String> logos, ImmutableList<String> colors) {
+    ProductDetectionData productDetectionData = ProductDetectionData.create(labels, logos, colors);
+    fakeProductDetection.setReturnValue(productDetectionData);
+
+    ProductPhotoDetector productPhotoDetector = new ProductPhotoDetector(fakeProductDetection);
```

Comment:
What do you think about making `ProductPhotoDetector productPhotoDetector ` a class variable (cause all of your methods are using a `ProductPhotoDetector` instance that are getting initialised separately anyways) ?

Reply:
Yes, I thought about it! At first, it was a class field and it was initialized in the setUp(). But, after adding the API class in the constructor, I decided to not have it as a field, as I have to create a new instance of the ProductPhotoDetector everytime, after I set the behaviour of of the FakeProductDetectionAPIImpl